File: Famicoco/app/src/main/java/com/example/mito/famicoco/PETFragment.java
package com.example.mito.famicoco;

import android.annotation.SuppressLint;
import android.os.AsyncTask;
import android.os.Bundle;
import androidx.fragment.app.Fragment;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;

import org.json.JSONArray;
import org.json.JSONException;

import java.io.IOException;
import java.io.InputStream;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;

import static com.example.mito.famicoco.MainActivity.ServerUrl;
import static com.example.mito.famicoco.MainActivity.tlIcon;

public class PETFragment extends Fragment implements SwipeRefreshLayout.OnRefreshListener {

    private CustomAdapter adapter;

    @BindView(R.id.pet_listview)
    ListView listView;
    @BindView(R.id.swipe_pet)
    SwipeRefreshLayout swipeRefreshLayout;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_pet, container, false);
        ButterKnife.bind(this, v);

        swipeRefreshLayout.setColorSchemeResources(R.color.MainColorDark,
                R.color.MainColor, R.color.MainColorDark, R.color.MainColor);
        swipeRefreshLayout.setOnRefreshListener(this);
        return v;
    }

    @Override
    public void onActivityCreated(Bundle bundle) {
        super.onActivityCreated(bundle);
        ArrayList<CustomData> list = new ArrayList<>();
        adapter = new CustomAdapter(getContext(), 0, list);
    }

    public PETFragment() {
    }

    @Override
    public void onResume() {
        UpData();
        super.onResume();
    }

    public void UpData() {
        @SuppressLint("StaticFieldLeak") AsyncTask<URL, Void, ArrayList<String>> task = new AsyncTask<URL, Void, ArrayList<String>>() {
            @Override
            protected ArrayList<String> doInBackground(URL... values) {
                HttpURLConnection con = null;
//                ArrayList<TLUpdateItem> tlUpdateItems = new ArrayList<>();
                ArrayList<String> stringArrayList = new ArrayList<>();
                try {
                    // アクセス先URL
                    final URL url = values[0];
                    // ローカル処理
                    // コネクション取得
                    con = (HttpURLConnection) url.openConnection();
                    con.setDoInput(true);
                    con.connect();
                    InputStream in = con.getInputStream();
                    String readSt = HttpGetTask.readInputStream(in);

                    //// 配列を取得する場合
                    JSONArray jsonArray = new JSONArray(readSt);

                    int m = jsonArray.length();
                    for (int i = 0; i < m; i++) {
                        String text = jsonArray.getString(i);
                        stringArrayList.add(text);
//                        item.setComment(text);
//                        item.setIcon(tlIcon[5]);
//                        item.setTime();

//                        item = new TLUpdateItem(data);
//                        tlUpdateItems.add(item);
                    }
                } catch (JSONException | IOException e) {
                    e.printStackTrace();
                } finally {
                    if (con != null) {
                        // コネクションを切断
                        con.disconnect();
                    }
                }
                return stringArrayList;
            }

            @Override
            protected void onPostExecute(ArrayList<String> list_pet_string) {
                ArrayList<CustomData> list_pet = new ArrayList<>();
                int m = list_pet_string.size();
                for (int i = 0; i < m; i++) {
                    CustomData item = new CustomData();
                    item.setComment(list_pet_string.get(i));
                    item.setTime();
                    item.setIcon(tlIcon[5]);
                    list_pet.add(item);
                }
                adapter = new CustomAdapter(getContext(), 0, list_pet);
                listView.setAdapter(adapter);
                Log.d("hoge", "here");
            }
        };
        try {
            URL url = new URL(ServerUrl + "get_pet/");
            task.execute(url);
        } catch (MalformedURLException e) {
            e.printStackTrace();
        }
    }

    @Override
    public void onRefresh() {
        UpData();
        swipeRefreshLayout.setRefreshing(false);
    }
}

File: Famicoco/app/src/main/java/com/example/mito/famicoco/PETFragment.kt
package com.example.mito.famicoco

import android.annotation.SuppressLint
import android.os.AsyncTask
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ListView
import androidx.fragment.app.Fragment
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import butterknife.BindView
import butterknife.ButterKnife
import com.example.mito.famicoco.MainActivity.Companion.ServerUrl
import com.example.mito.famicoco.MainActivity.Companion.selectIconMapId
import com.example.mito.famicoco.MainActivity.Companion.tlIcon
import org.json.JSONArray
import java.net.HttpURLConnection
import java.net.MalformedURLException
import java.net.URL
import java.util.ArrayList

class PETFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener {
    
    private lateinit var adapter: CustomAdapter
    
    @BindView(R.id.pet_listview)
    internal lateinit var listView: ListView
    @BindView(R.id.swipe_pet)
    internal lateinit var swipeRefreshLayout: SwipeRefreshLayout
    
    override fun onCreateView(
            inflater: LayoutInflater,
            container: ViewGroup?,
            savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(
                R.layout.fragment_pet,
                container,
                false
        )
        ButterKnife.bind(
                this,
                view
        )
        
        swipeRefreshLayout.setColorSchemeResources(
                R.color.MainColorDark,
                R.color.MainColor, R.color.MainColorDark,
                R.color.MainColor
        )
        swipeRefreshLayout.setOnRefreshListener(this)
        return view
    }
    
    override fun onActivityCreated(bundle: Bundle?) {
        super.onActivityCreated(bundle)
        val list = ArrayList<CustomData>()
        adapter = CustomAdapter(
                context,
                0,
                list
        )
    }
    
    override fun onResume() {
        update()
        super.onResume()
    }
    
    private fun update() {
        selectIconMapId = HashMap()
        @SuppressLint("StaticFieldLeak")
        val task = object : AsyncTask<URL, Void, ArrayList<String>>() {
            override fun doInBackground(vararg values: URL): ArrayList<String> {
                var con: HttpURLConnection? = null
                //                ArrayList<TLUpdateItem> tlUpdateItems = new ArrayList<>();
                val stringArrayList = ArrayList<String>()
                try {
                    // アクセス先URL
                    val url = values[0]
                    // ローカル処理
                    // コネクション取得
                    con = url.openConnection() as HttpURLConnection
                    con.doInput = true
                    con.connect()
                    val `in` = con.inputStream
                    val readSt = HttpGetTask.readInputStream(`in`)
                    
                    // // 配列を取得する場合
                    val jsonArray = JSONArray(readSt)
                    
                    val m = jsonArray.length()
                    for (i in 0 until m) {
                        val text = jsonArray.getString(i)
                        stringArrayList.add(text)
                        //                        item.setComment(text);
                        //                        item.setIcon(tlIcon[5]);
                        //                        item.setTime();
                        
                        //                        item = new TLUpdateItem(data);
                        //                        tlUpdateItems.add(item);
                    }
                } catch (e: Exception) {
                    e.printStackTrace()
                } finally {
                    con?.disconnect()
                }
                return stringArrayList
            }
            
            override fun onPostExecute(list_pet_string: ArrayList<String>) {
                val petList = ArrayList<CustomData>()
                val m = list_pet_string.size
                for (i in 0 until m) {
                    val item = CustomData()
                    item.also {
                        it.comment = list_pet_string[i]
                        it.setTime()
                        it.icon = tlIcon[5]
                    }
                    petList.add(item)
                }
                adapter = CustomAdapter(
                        context,
                        0,
                        petList
                )
                listView.adapter = adapter
            }
        }
        try {
            val url = URL(ServerUrl + "get_pet/")
            task.execute(url)
        } catch (e: MalformedURLException) {
            e.printStackTrace()
        }
    }
    
    override fun onRefresh() {
        update()
        swipeRefreshLayout.isRefreshing = false
    }
}
